packages/quantic/force-app/main/default/lwc/quanticResultAction/quanticResultAction.js

```diff
@@ -0,0 +1,149 @@
+import {LightningElement, api} from 'lwc';
+
+/** @typedef {import("coveo").Result} Result */
+
+/**
+ * The `QuanticResultAction` component allows the end user to perform a specific action on a result.
+ * @category Search
+ * @category Insight Panel
+ * @example
+ * <c-quantic-result-action result={result} event-name={attachtocase} icon-name="utility:chat" label="Attach result" selected label-when-on="Detach result"></c-quantic-result-action>
+ */
+export default class QuanticResultAction extends LightningElement {
+  /**
+   * The label displayed in the tooltip of the result action button.
+   * @api
+   * @type {string}
+   */
+  @api label;
+  /**
+   * The label displayed in the tooltip when the result action button is selected.
+   * @api
+   * @type {string}
+   */
+  @api labelWhenOn;
```

Comment:
```suggestion
  @api selectedLabel;
```
To tie it to the `selected` prop.